File: 2025Irrgarten/src/Irrgarten/Dice.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package Irrgarten;
import java.util.Random;
/**
 *
 * @author astralisx
 */
public class Dice {
    static private final int MAX_USES=5;
    static private final float MAX_INTELLIGENCE=10.0f;
    static private final float MAX_STRENGTH=10.0f;
    static private final float RESURRECT_PROB=0.3f;
    static private final int WEAPONS_REWARD = 2;
    static private final int SHIELD_REWARD = 3;
    static private final int HELATH_REWARD=5;
    static private final int MAX_ATTACK = 3;
    static private final int MAX_SHIELD = 2;
    static private final Random generator = new Random();
    
    public static int randomPos(int max){
        return generator.nextInt(0,max);
    }
    
    public static int whoStarts(int players){
        return generator.nextInt(0,players);
    }
    
    public static float randomIntelligence(){
        return generator.nextFloat() * MAX_INTELLIGENCE;
    }
    
    public static float randomStrength(){
        return generator.nextFloat() * MAX_STRENGTH;
    }
    
    public static boolean resurrectPlayer(){
       return generator.nextFloat() <= RESURRECT_PROB;
    }
    
    public static int weaponsReward(){
        return generator.nextInt(WEAPONS_REWARD+1);
    }

    public static int shieldReward(){
        return generator.nextInt(SHIELD_REWARD+1);
    }

    public static int healthReward(){
        return generator.nextInt(HELATH_REWARD)+1;
    }

    public static float weaponPower(){
        return generator.nextFloat()*MAX_ATTACK;
    }

    public static float shieldPower(){
        return generator.nextFloat()*MAX_SHIELD;
    }

    public static int usesLeft(){
        return generator.nextInt(MAX_USES+1);
    }

    public static float intensity(float competence){
        return generator.nextFloat()*competence;
    }

    public static boolean discardElement(int usesLeft){
        float prob = (float)usesLeft/MAX_USES;
        return generator.nextFloat() >= prob;
    }



}   

File: IrrgartenRuby/dice.rb
# encoding: UTF-8
module Irrgarten
    class Dice
        @@MAX_USES = 5
        @@MAX_INTELLIGENCE = 10.0
        @@MAX_STRENGTH = 10.0
        @@RESURRECT_PROB = 0.3
        @@WEAPONS_REWARD = 2
        @@SHIELDS_REWARD = 3
        @@HEALTH_REWARD = 5
        @@MAX_ATTACK = 3
        @@MAX_SHIELD = 2

        @@generator = Random.new

        def self.random_pos(max)
            @@generator.rand(max.to_i)
        end

        def self.who_starts(nplayers)
            @@generator.rand(nplayers.to_i)
        end
        #Nunca se toma el valor escrito como parametro, es por este motivo
        #que se suma uno en los reward. En el caso de no escribir to_f, nos
        #devolveria un int porque el valor de los objetos de clase son enteros
        def self.random_intelligence
            @@generator.rand(@@MAX_INTELLIGENCE.to_f)
        end

        def self.random_strength
            @@generator.rand(@@MAX_STRENGTH.to_f)
        end

        def self.resurrect_player
            (@@generator.rand <= @@RESURRECT_PROB)
        end

        def self.weapons_reward
            @@generator.rand(@@WEAPONS_REWARD.to_i + 1)
        end

        def self.shields_reward
            @@generator.rand(@@SHIELDS_REWARD.to_i + 1)
        end

        def self.health_reward
            @@generator.rand(@@HEALTH_REWARD.to_i + 1)
        end

        def self.weapon_power
            @@generator.rand(@@MAX_ATTACK.to_f)
        end

        def self.shield_power
            @@generator.rand(@@MAX_SHIELD.to_f)
        end

        def self.uses_left
            @@generator.rand(@@MAX_USES.to_i + 1)
        end

        def self.intensity(competence)
            @@generator.rand(competence.to_f)
        end

        def self.dicard_element(uses_left)
            probabilidad = uses_left.to_f / @@MAX_USES.to_f
            (@@generator.rand >= probabilidad)
        end
    end
end
